### ISSUE TITLE
Contribute to VSCode directly?

### ISSUE THREAD

**Issue by supervanya** [donebd/vscode-keypromoter#50]
Making this as a plugin seems very cumbersome, what do you think about making this available as core of the vscode?

I started poking around the vscode codebase, wondering if you want to help out https://github.com/supervanya/vscode-key-tutor

**Comment by donebd** [comment]
I have already researched this topic, and there was already a similar attempt at implementation and it was removed by reasons:
https://github.com/microsoft/vscode/issues/78091

**Comment by walles** [comment]
From <https://github.com/microsoft/vscode/issues/78091#issuecomment-521640092>:

> we are open to accept features like the "keybindings teacher"

So upstream is interested, or at least not explicitly hostile to this.

**Comment by supervanya** [comment]
> So upstream is interested, or at least not explicitly hostile to this.

Yeah that is what got me interested. But the codebase is massive and I couldn't figure out how to hook into the right events to see their source (like I want only pointer events)
